framework/src/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -698,6 +698,21 @@ private Pattern getOnlyDefsPattern(Map<String, String> options) {
         return swString.split(",");
     }
 
+    @Override
+    public void warn(String format, Object... args) {
+        String msg = String.format(format, args);
+        this.messager.printMessage(Kind.MANDATORY_WARNING, msg);
```

Comment:
@wmdietl  These warnings need to be suppresable.  Any ideas on how best to implement that?

Reply:
I think if you attach the message to an `Element` as source, you can use the normal `@SuppressWarnings` to achieve this.